Sources/VSM/Documentation.docc/ComprehensiveGuide/ModelDefinition.md

```diff
@@ -22,142 +22,141 @@ First, we'll take a look at the shape of the `LoadUserProfileViewState` and its
 
 ```swift
 enum LoadUserProfileViewState {
-    case initialized(LoaderModel)
+    case initialized(LoaderModeling)
     case loading
-    case loadingError(ErrorModel)
+    case loadingError(LoadingErrorModeling)
     case loaded(UserData)
+}
 
-    struct LoaderModel {
-        let load: () -> AnyPublisher<LoadUserProfileViewState, Never>
-    }
+protocol LoaderModeling {
+    func load() -> AnyPublisher<LoadUserProfileViewState, Never>
+}
 
-    struct ErrorModel {
-        let message: String
-        let retry: () -> AnyPublisher<LoadUserProfileViewState, Never>
-    }
+protocol LoadingErrorModeling {
+    let message: String
+    func retry() -> AnyPublisher<LoadUserProfileViewState, Never>
 }
 ```
 
-After we have a good idea of how the states and data should flow, we can begin implementing the model by extending the corresponding model struct definition. Our entire implementation for this model will live in this struct extension.
+After we have a good idea of how the states and data should flow, we can begin implementing the model by creating a struct that implements the corresponding model protocol. Our entire implementation for this model will live in this struct.
 
 ```swift
-extension LoadUserProfileViewState.LoaderModel {
+struct LoaderModel: LoaderModeling {
     ...
 }
 ```
 
-Next, we'll create an initializer that defines the behavior of the `load()` action property on the struct.
+Next, we'll create an initializer that collects the data required for the `load()` action.
 
 ```swift
-extension LoadUserProfileViewState.LoaderModel {
+struct LoaderModel: LoaderModeling {
+    let userId: Int
+
     init(userId: Int) {
-        load = {
-            ...
-        }
+        self.userId = userId
     }
```

Comment:
Yep! I debated this very thing in my head when writing it. I felt like it would be helpful for some less experienced engineers to see it spelled out and realize that they can perform custom logic within a custom initializer if needed.

Reply:
There are other examples in your docs now that do include initializers doing custom work (not just member initialization). So maybe it would be ok to remove init's like this one, since you do have examples of custom init elsewhere?

Reply:
I would agree with preferring to removing the initializers that aren't needed.  I would argue that knowing that memberwise initializers are autogenerated for structs is Structs 101.  So unless you are brand new to Swift, I think it would be safe to assume the reader understands this.  I believe even Apple's beginner SwiftUI tutorials just use the autogenerated inits.

Reply:
Sounds good! I removed the explicit initializer for the loader examples and modified the text a little.